src/core.jl

```diff
@@ -1,3 +1,51 @@
+#################################################
+# Test mode
+# This controls how test cases are handled
+abstract type TestMode end
+struct InteractiveMode <: TestMode end
+struct NonInteractiveMode <: TestMode end
+
+"""
+Predefined CI environment variables
+
+# References
+
+* Travis: https://docs.travis-ci.com/user/environment-variables/#default-environment-variables
+* Appveyor: https://www.appveyor.com/docs/environment-variables/
+"""
+const CI_ENVIRONMENTS = Dict(
+    "CI" => "true",
+    "APPVEYOR" => "true",
+    "TRAVIS" => "true",
+    "CONTINUOUS_INTEGRATION" => "true",
+    "DEBIAN_FRONTEND" => "noninteractive"
+)
+
+function TESTMODE()
+    global GLOBAL_TESTMODE
+    if !isdefined(ReferenceTests, :GLOBAL_TESTMODE)
+        # it's only called once in runtime
+        GLOBAL_TESTMODE = _get_testmode()
+    end
+    return GLOBAL_TESTMODE
+end
```

Comment:
`TestMode()` generates a runtime singleton flag.
Please enlighten me if there's a best-practice as an alternative to this usage.